packages/toolkit/src/query/utils/joinUrls.ts

```diff
@@ -18,8 +18,9 @@ export function joinUrls(
     return url
   }
 
+  const delimiter = base.endsWith('/') || !url.startsWith('?') ? '/' : ''
```

Comment:
Can we refactor this so it also handle cases where the base ends with `@`?

**Example 1:**
```
const delimeter = (base.endsWith('?') || base.endsWith('@')) ? '' : '/'
```

**Example 2**
```
let delimeter = '/'
if (base.endsWith('?') || base.endsWith('@')) {
  delimeter = ''
}
```

Reply:
This is still meant to work for normal http-style urls. If we added a myriad of symbols here, it would just become weirder and weirder. Do you maybe use too much of a string for the `baseUrl`? Usually you would already cut that off at the domain.

Reply:
> This is still meant to work for normal http-style urls. If we added a myriad of symbols here, it would just become weirder and weirder. Do you maybe use too much of a string for the `baseUrl`? Usually you would already cut that off at the domain.

It's an external endpoint I'm using which I have no power of, unfortunately.
I have to append a dynamic value after `@` and that's the reason the current solution doesn't work for my particular case.

I can make a workaround by hardcoding a random value - but that is **not** a long-term solution.

Reply:
Assume this is your url:

`https://example.org/some/sub/path@foo`

Then in your case, you would have to use a `baseUrl` of `https://example.org/some/sub` and have every endpoint be declared as `path@foo`, `path@bar` etc.

Reply:
That is another "hack" you can do,  but still not a robust solution. 



> Assume this is your url:
> 
> `https://example.org/some/sub/path@foo`
> 
> Then in your case, you would have to use a `baseUrl` of `https://example.org/some/sub` and have every endpoint be declared as `path@foo`, `path@bar` etc.

`path` is and will always be a base-url, and **not** an endpoint. I do have to append on **_every_** endpoint which is nothing but _code duplication_. 

This is why I think redux-toolkit should handle this scenario.

Reply:
It's simple: urls in the internet don't work like that. Joining urls or paths doesn't work like that. The path separator for urls is `/`. 
If we add that `@` here, it will cause bugs for others because it is not part of any standard. And five minutes down the road we will add even more symbols and cause even more bugs for even more people.

You won't get around adding that yourself or writing a helper for yourself - this is a non-standard scenario and we cannot support it.

Reply:
> ...five minutes down the road we will add even more symbols and cause even more bugs for even more people.
> 

That's why I suggest having an optional method (or something) to customise the url based on the user's need. That would solve everything now and later on.

Reply:
You can just handle that in a custom `baseQuery`:

```js
const myCustomBase = 'http://foo.bar/asd@
const defaultBq = fetchBaseQuery({})
const myBaseQuery = (arg, ...rest) => {
  const newArg = typeof arg == 'string' ? myCustomBase + arg : { ...arg, url: myCustomBase+arg.url }
  return defaultBq(newArg, ...rest)
}
```